dynamic load test fails on master

Just cloned and repeated the steps in README.

```
$ make test
Running tests...
Test project /home/ror6ax/Development/opentracing-c/build
    Start 1: dynamic_load_test
1/2 Test #1: dynamic_load_test ................***Exception: Child aborted  0.23 sec
    Start 2: tracer_test
2/2 Test #2: tracer_test ......................   Passed    0.00 sec

50% tests passed, 1 tests failed out of 2

Total Test time (real) =   0.24 sec

The following tests FAILED:
	  1 - dynamic_load_test (Child aborted)
Errors while running CTest
make: *** [Makefile:107: test] Error 8
```

Thanks.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -105,9 +105,13 @@ if(OPENTRACINGC_LINT)
 endif()
 
 if(BUILD_TESTING)
-  set(test_src
-    "test/dynamic_load_test.c"
-    "test/tracer_test.c")
+  set(test_src "test/tracer_test.c")
+  if(BUILD_SHARED_LIBS AND OPENTRACINGC_HAVE_WEAK_SYMBOLS)
+    set(build_dynamic_load_test ON)
+  endif()
+  if(build_dynamic_load_test)
+    list(APPEND test_src "test/dynamic_load_test.c")
+  endif()
 
   foreach(test_case_src ${test_src})
     get_filename_component(test_component ${test_case_src} NAME_WE)
@@ -117,7 +121,7 @@ if(BUILD_TESTING)
     list(APPEND test_executables ${test_component})
   endforeach()
 
-  if(OPENTRACINGC_HAVE_WEAK_SYMBOLS)
+  if(build_dynamic_load_test)
     add_library(mock_tracing_lib SHARED "test/mock_tracing_lib.c")
     target_compile_definitions(mock_tracing_lib PRIVATE DEFINE_HOOK)
     target_link_libraries(mock_tracing_lib PUBLIC opentracingc)

--- a/scripts/build.sh
+++ b/scripts/build.sh
@@ -51,7 +51,11 @@ function main() {
     local prefix_arg
     prefix_arg=$1
     prefix_arg=${prefix_arg:+"-DCMAKE_INSTALL_PREFIX=$prefix_arg"}
-    cmake -DCMAKE_BUILD_TYPE=Debug ${prefix_arg} -DOPENTRACINGC_COVERAGE=${COVERAGE:OFF} ..
+    cmake -DCMAKE_BUILD_TYPE=Debug \
+        ${prefix_arg} \
+        -DBUILD_SHARED_LIBS=ON \
+        -DOPENTRACINGC_COVERAGE=${COVERAGE:OFF} \
+        ..
 
     if make all -j3; then
         if valgrind_check ./tracer_test && \

